src/shared/hooks/useAsynCall.ts

```diff
@@ -0,0 +1,47 @@
+import * as React from 'react';
```

Comment:
We have tow of these files ?

Reply:
one is a memoized callback, it's used for different purposes. there is `useAsyncCall` and `useAsyncCallback`

Reply:
Ohh, my bad. I suggest we change the name to make the purpose explicit. At present, the names looks very similar.